marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -469,6 +529,30 @@ handleAction settings (OpenLoginPopup intendedAction) = do
     Anonymous -> pure unit
     GithubUser -> handleAction settings intendedAction
 
+handleAction settings (ConfirmUnsavedNavigationAction intendedAction modalAction) = do
+  toConfirmUnsavedNavigation intendedAction $ ConfirmUnsavedNavigation.handleAction settings modalAction
+  handleAction settings CloseModal
+  case modalAction of
+    ConfirmUnsavedNavigation.Cancel -> pure unit
+    ConfirmUnsavedNavigation.DontSaveProject -> do
+      liftEffect $ Console.log "DontSaveProject"
+      -- FIXME: this currently is never ending loop, need to comunicate
+      -- with the preventNavigation that I want to do the action anyway
+      -- The only way I can think of at the time is to put a boolean in the
+      -- state
+      handleAction settings intendedAction
+    ConfirmUnsavedNavigation.SaveProject -> do
+      liftEffect $ Console.log "SaveProject"
+      state <- H.get
+      -- FIXME: This was taken from the view, from the gistModal helper. I think we should
+      -- refactor into a Save Action that does this check and receives a Maybe Action as
+      -- a continuation
```

Comment:
Sounds good, I'd like to see the refactoring though, I can't think how it would look

Reply:
I'll change the FIXME for a TODO and do it in a separate PR